packages/datatrak-web/src/api/mutations/useSubmitSurvey.ts

```diff
@@ -50,14 +50,19 @@ export const useSubmitSurvey = () => {
         return;
       }
 
-      await post('submitSurvey', {
+      return await post('submitSurvey', {
         data: { ...surveyResponseData, answers },
       });
     },
     {
-      onSuccess: () => {
+      onSuccess: data => {
         successToast("Congratulations! You've earned a coconut", Coconut);
-        navigate(generatePath(ROUTES.SURVEY_SUCCESS, params));
+        // include the survey response data in the location state, so that we can use it to generate QR codes
+        navigate(generatePath(ROUTES.SURVEY_SUCCESS, params), {
+          state: {
+            surveyResponse: JSON.stringify(data),
```

Comment:
This could just be an object and we could avoid stingify and parse right?

Reply:
No, with react router, you can't pass complex objects as location state, they have to be stringified

Reply:
I reckon it does work as it's just using the browser [history object](https://developer.mozilla.org/en-US/docs/Web/API/History_API/Working_with_the_History_API) (plus I tested it :)) but I don't mind too much either way 👍